src/ui/components/DataTable/CustomizeColumns.tsx

```diff
@@ -1,134 +1,151 @@
+
+	useEffect(() => {
+		const nextColumns = [...columns].map(c => ({
+			...c,
+			hidden: !config.columns.some(col => col.key === c.key),
+		}));
+		setColumns(nextColumns);
+	}, [config]);
+
+	const onChange = (key: string) => () => {
+		const nextColumns = [...columns];
+		const i = nextColumns.findIndex(c => c.key === key);
+		if (i !== -1) {
+			nextColumns[i] = { ...nextColumns[i] };
+			nextColumns[i].hidden = !nextColumns[i].hidden;
+			setColumns(nextColumns);
+		}
+	};
+
+	const reset = () => setColumns(initialColumns);
+
+	const exit = () => {
+		reset();
+		onHide();
+	};
+
+	const restore = async () => {
+		await toWorker("main", "updateColumns", {
+			columns: config.fallback,
+			key: config.tableName,
+		});
+		onHide();
+		onSave();
+	};
+
+	const save = async () => {
+		const enabledColumns: string[] = columns
+				.filter(c => !c.hidden)
+				.map(c => c.key),
+			currentColumns: string[] = config.columns.map(c => c.key);
+
+		// Find columns we need to remove, and columns we need to add
+		const removeColumns: string[] = difference(currentColumns, enabledColumns),
+			addColumns: string[] = difference(enabledColumns, currentColumns);
+
+		// Apply removals and adding to currentColumns while trying to preserve the order of currentColumns
+		const nextColumns: string[] = currentColumns.filter(
+			c => !removeColumns.includes(c),
+		);
+		nextColumns.push(...addColumns);
+
+		await toWorker("main", "updateColumns", {
+			columns: nextColumns,
+			key: config.tableName,
+		});
+		onHide();
+		onSave();
+	};
+
+	const colsGrouped: { [key: string]: ColConfig[] } = groupBy(
+		columns,
+		c => c.cat,
+	);
 
 	return (
-		<Modal animation={false} centered show={show} onHide={onHide}>
-			<Modal.Header closeButton>Customize Columns</Modal.Header>
+		<Modal
+			animation={false}
+			size={"lg"}
+			show={show}
+			onEscapeKeyDown={exit}
+			centered
+		>
+			<Modal.Header onHide={exit} closeButton>
+				Customize Columns
+			</Modal.Header>
 			<Modal.Body>
-				<p>
-					Click and drag to reorder columns, or use the checkboxes to show/hide
-					columns.
-				</p>
-				{hasSuperCols ? (
-					<p className="text-danger">
-						This table has two header rows. That means you can enable/disable
-						columns, but not reorder them.
-					</p>
-				) : null}
-				<Container
-					helperClass="sort-inside-modal"
-					isDragged={isDragged}
-					onSortStart={() => {
-						setIsDragged(true);
-					}}
-					onSortEnd={args => {
-						setIsDragged(false);
-						if (!hasSuperCols) {
-							onSortEnd(args);
-						}
-					}}
-				>
-					{colOrder.map(({ colIndex, hidden }, i) => {
-						const col = cols[colIndex];
-						return (
-							<Item
-								key={colIndex}
-								index={i}
-								onToggleHidden={onToggleHidden(i)}
-								hidden={hidden}
-								col={col}
-							/>
-						);
-					})}
-				</Container>
+				<ul className="list-group list-group-flush">
```

Comment:
![image](https://user-images.githubusercontent.com/45333542/150992989-fd222966-4030-40e6-9f75-5c2e3bb6e33c.png)
ended up doing p-0 otherwise there was a weird gap, but yes it is much better